internal/ingress/controller/parser/parser.go

```diff
@@ -590,6 +623,70 @@ func overrideRoute(route *Route,
 	}
 }
 
+// normalizeProtocols prevents users from mismatching grpc/http
+func normalizeProtocols(route *Route) {
+	protocols := route.Protocols
+	var http, grpc bool
+
+	for _, protocol := range protocols {
+		if strings.Contains(*protocol, "grpc") {
+			grpc = true
+		}
+		if strings.Contains(*protocol, "http") {
+			http = true
+		}
+		if sanitizeProtocol(*protocol) != true {
+			http = true
+		}
+	}
+
+	if grpc && http {
+		route.Protocols = kong.StringSlice("http", "https")
+	}
+	// length check
+	if len(route.Protocols) < 1 {
+		return
+	}
+}
+
+// sanitizeProtocol returns a bool of whether string is a valid protocol
+func sanitizeProtocol(protocol string) bool {
+	re2 := regexp.MustCompile(`\Ahttps$|\Ahttp$|\Agrpc$|\Agrpcs$`)
+	match := re2.MatchString(protocol)
+	return match
+}
+
+// overrideRouteByAnnotation sets Route protocols via annotation
+func overrideRouteByAnnotation(route *Route, anns map[string]string) {
+	if anns == nil {
+		return
+	}
+	protocols := annotations.ExtractProtocolNames(anns)
+	var prots []*string
+	for _, prot := range protocols {
+		prots = append(prots, kong.String(prot))
+	}
+	route.Protocols = prots
```

Comment:
This function also needs to validate protocols since a user can input any random string as the value for the annotation.